src/Client/Connection.cpp

```diff
@@ -109,6 +109,9 @@ void Connection::connect(const ConnectionTimeouts & timeouts)
         }
 
         in = std::make_shared<ReadBufferFromPocoSocket>(*socket);
+        if (async_callback)
+            in->setAsyncCallback(std::move(async_callback));
```

Comment:
That looks quite strange.
We set callback only when `connect` is called. But
* Connection can be reused. But old callback will be attached to socket.
* It is not clear why callback is set before first `connect` call (especially if connection was reused).

Reply:
Ok, I see that `setAsyncCallback` sets it also to socket. Probably, it's ok.

Reply:
Yes, you're right, it's strange. I made it when tried use fibers for the first time and forgot to change this place, I'll fix it